utils/avx2_calls.h

```diff
@@ -179,6 +183,10 @@ extern "C" {
 #define AVX2_STREAMING_STORE_FLOATS(X,Y)   _mm256_stream_pd(X,Y)
 #define AVX2_STREAMING_STORE_INTS(X,Y)     _mm_stream_si128(X,Y)
 
+/* returns Z + XY*/
+#define AVX2_FMA_ADD_FLOATS(X,Y,Z)          _mm256_fmadd_pd(X,Y,Z)
+#define AVX2_FMA_ADD_TRUNCATE_FLOATS(X,Y,Z) _mm256_round_pd(_mm256_fmadd_pd(X,Y,Z),_MM_FROUND_TO_ZERO|_MM_FROUND_NO_EXC)
```

Comment:
Is this being used anywhere? `round_pd/ps` returns a float, not an int, right? So it's not useful for the bin index?

Reply:
Good catch, AVX2_FMA_ADD_TRUNCATE_FLOATS specifically is not used, indeed! (but e.g. AVX_FMA_ADD_TRUNCATE_FLOATS is useful in DDrppi - to take the floor value of the rp index before adding in the pi index)

Reply:
Oh, right; I forgot that part of the code used floats to compute the integer index. @manodeep I guess the reason it's done this way is to be able to use FMA?

Reply:
Originally I had the cast-to-float to perform some operations (mult? truncate?) that there not available for integer vectors. The question might be whether using FMA here helps. 

@adematti You might be able to combine the two ops into one with something from this [list](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=fmadd_round_pd&expand=3191,3675,3737,3192). `_mm512_fmadd_round_pd` would be the exact replacement for what you already have, but you will likely need to use the masked variant